src/components/AccountManagement.vue

```diff
@@ -0,0 +1,161 @@
+<script setup lang="ts">
+import { getApp } from 'firebase/app';
+import {
+  deleteUser,
+  getAuth,
+  onAuthStateChanged,
+  signInWithEmailAndPassword,
+  updateProfile,
+  User,
+} from 'firebase/auth';
+import {
+  doc,
+  deleteDoc,
+  getDoc,
+  getFirestore,
+  setDoc,
+} from 'firebase/firestore';
+import { onBeforeUnmount, ref } from 'vue';
+
+import { router } from '../router';
+
+const auth = getAuth();
+const db = getFirestore(getApp());
+
+// Fields
+const email = ref();
+const firstName = ref();
+const lastName = ref();
+const reloginPassword = ref();
+
+// Toggles
+const editingName = ref(false);
+const showRelogin = ref(false);
+
+async function getUserInfo(user: User) {
+  if (email.value) {
```

Comment:
Is it possible the email is loaded without the first name / last name being loaded?

Reply:
That shouldn't be possible. But on second thought, this guard shouldn't be necessary anyway since getUserInfo is only called onAuthStateChanged, and that only happens when loading and unloading this view.